core/include/traccc/edm/cluster.hpp

```diff
@@ -9,31 +9,46 @@
 
 // Project include(s).
 #include "traccc/edm/cell.hpp"
+#include "traccc/geometry/pixel_data.hpp"
 
 // System include(s).
 #include <functional>
 #include <vector>
 
 namespace traccc {
 
-using position_estimation = std::function<vector2(channel_id, channel_id)>;
-using signal_modeling = std::function<scalar(scalar)>;
+/// Function for signal modelling
+inline scalar signal_cell_modelling(scalar signal_in) {
+    return signal_in;
+}
+
+/// Function for pixel segmentation
+inline vector2 position_from_cell(cell c, pixel_data pixel, bool is_default) {
+    if (is_default)
+        return {static_cast<scalar>(c.channel0),
+                static_cast<scalar>(c.channel1)};
+    else
+        // Retrieve the specific values based on module idx (for now hard-coded)
+        return {pixel.min_center_x + c.channel0 * pixel.pitch_x,
+                pixel.min_center_y + c.channel1 * pixel.pitch_y};
+}
 
+inline vector2 get_pitch(pixel_data pixel) {
+    // return the values based on the module idx (for now hard-coded)
+    return {pixel.pitch_x, pixel.pitch_y};
+}
+
+using position_estimation = std::function<vector2(channel_id, channel_id)>;
 struct cluster_id {
 
     event_id event = 0;
+    scalar module_idx = 0;
```

Comment:
Please change this to `std::size_t`. A bit confusingly `scalar` refers to a floating point number. Either `double` or `float` in our current builds.